lib/rbs_rails/active_record.rb

```diff
@@ -499,10 +499,12 @@ def authenticate_#{attribute}: (::String) -> (#{klass_name} | false)
                            sql_type_to_class(col.type)
                          end
           end
+          sql_class_name = col.type == :datetime ? '::Time' : sql_type_to_class(col.type)
```

Comment:
```rb
User.take.created_at_for_database.class
#=> Time

Reply:
The `sql_type_to_class` method already includes a branch for `:datetime`, which maps to `ActiveSupport::TimeWithZone`.
https://github.com/pocke/rbs_rails/blob/6f48df9b66fec08b9e5da676c7f4af563d5787ac/lib/rbs_rails/active_record.rb#L582-L583
However, in the case of `_before_type_cast` and `_for_database`, `:datetime` values are instances of the `Time` class.